src/test/java/bio/terra/service/filedata/DrsServiceTest.java

```diff
@@ -651,14 +667,91 @@ public void postAccessUrlForObjectIdInvalidPassport() {
         UnauthorizedException.class,
         () ->
             drsService.postAccessUrlForObjectId(
-                googleDrsObjectId, "gcp-passport-us-central1", drsPassportRequestModel));
+                googleDrsObjectId, "gcp-passport-us-central1", drsPassportRequestModel, null));
+  }
+
+  @Test
+  public void testSignGcpUrlNoProject() {
+    testSignGcpUrl(null, false);
+  }
+
+  @Test
+  public void testSignGcpUrlSpecifiedProject() {
+    testSignGcpUrl("myProject", false);
+  }
+
+  @Test
+  public void testSignGcpUrlSpecifiedProjectSelfHosted() {
+    testSignGcpUrl("myProject", true);
+  }
+
+  public void testSignGcpUrl(String billingProject, boolean selfHostedDataset) {
```

Comment:
This is doing a pretty similar thing, but with JUnit 5 you could rewrite this as a parameterized test.

```
  private static Stream<Arguments> testSignGcpUrl() {
    return Stream.of(
      arguments(null, false),
      arguments("myProject", false),
      arguments("myProject", true));
  }
  
  @ParameterizedTest
  @MethodSource
  void testSignGcpUrl(String billingProject, boolean selfHostedDataset) {
```

Reply:
Oh cool, lemme try that